### PR TITLE
stringify only when necessary

### PR DESCRIPTION
**Pull request by codingsince1985**
Benchmark setting span and transaction context.
Before
```
$ go test -bench=.
goos: linux
goarch: amd64
pkg: go.elastic.co/apm/module/apmot
BenchmarkSetSpanContext-4          	  300000	      4201 ns/op
BenchmarkSetTransactionContext-4   	  300000	      4593 ns/op
PASS
ok  	go.elastic.co/apm/module/apmot	2.881s
```
After
```
$ go test -bench=.
goos: linux
goarch: amd64
pkg: go.elastic.co/apm/module/apmot
BenchmarkSetSpanContext-4          	 1000000	      2148 ns/op
BenchmarkSetTransactionContext-4   	  500000	      2297 ns/op
PASS
ok  	go.elastic.co/apm/module/apmot	3.803s
```

### CODE REVIEW

File: module/apmot/span_test.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
+package apmot
```
Comment (axw) [inline module/apmot/span_test.go]: This file is failing the `goimports` check in CI, as the imports aren't grouped. Also, it'll fail the license check, as it's missing a header. You can fix these two things by running `make update-licenses fmt`

File: module/apmot/span_test.go [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+import (
+	"go.elastic.co/apm"
+	"testing"
+)
+
+func BenchmarkSetSpanContext(b *testing.B) {
```
Comment (axw) [inline module/apmot/span_test.go]: I think these benchmarks are a bit too low level, seeing as it's only possible for `setSpanContext` and `setTransactionContext` to be called via `Span.Finish`/`FinishWithOptions`.

How about changing them to start and finish real OT spans? i.e.

```go
tracer := apmot.New(apmot.WithTracer(apmtest.DiscardTracer))
for i := 0; i < b.N; i++ {
    span := tracer.StartSpanWithOptions("span", opentracing.StartSpanOptions{
        Tags: map[string]interface{}{/* tags go here */},
    })
    span.Finish()
}
```

Comment (codingsince1985) [reply module/apmot/span_test.go]: Unfortunately, `tracer` in this case is of `opentracing.Tracer` type, which doesn't have `StartSpanWithOptions()`, I may have to go with `StartSpan()`. Will look into it tomorrow.

Comment (axw) [reply module/apmot/span_test.go]: Sorry, I was conflating the implementation with the interface. You could instead pass an `opentracing.Tags` into `tracer.StartSpan`.

Comment (codingsince1985) [reply module/apmot/span_test.go]: Benchmark span finish.
Before
```
$ go test -bench=. tracer_test.go
goos: linux
goarch: amd64
pkg: go.elastic.co/apm/module/apmot
BenchmarkSpanSetSpanContext-4   	  200000	     10252 ns/op
BenchmarkSpanSetTxContext-4     	  100000	     11307 ns/op
PASS
ok  	go.elastic.co/apm/module/apmot	3.552s
```
After
```
$ go test -bench=. tracer_test.go
goos: linux
goarch: amd64
pkg: go.elastic.co/apm/module/apmot
BenchmarkSpanSetSpanContext-4   	  200000	      7656 ns/op
BenchmarkSpanSetTxContext-4     	  200000	      8064 ns/op
PASS
ok  	go.elastic.co/apm/module/apmot	3.480s
```

Comment (axw) [reply module/apmot/span_test.go]: Nice, thanks! Will you delete span_test.go? I think the benchmarks you added to tracer_test.go are sufficient?

Comment (codingsince1985) [reply module/apmot/span_test.go]: done.

**Comment by axw** [comment]
jenkins run the tests please

**Pull request merged**